components/director/internal/domain/label/repository.go

```diff
@@ -174,6 +176,59 @@ func (r *repository) DeleteByKey(ctx context.Context, tenant string, key string)
 	return nil
 }
 
+func (r *repository) GetRuntimesIDsWhereLabelsMatchSelector(ctx context.Context, tenantID, selectorKey, selectorValue string) ([]string, error) {
```

Comment:
It some ways, yes, because in scenario assigments we use `selectors`. I would left it as it is and rename to `GetRuntimesIDsByKeyAndValue` and also rename the parameters.